File: code/C++/error_handling.cpp
// C++: error handling via exceptions
#include <iostream>
#include <stdexcept>

int parse_positive(int x) {
    if (x <= 0)
        throw std::runtime_error("input must be positive");
    return x;
}

int main() {
    try {
        int v = parse_positive(-3);
        std::cout << "ok: " << v << "\n";
    } catch (const std::exception& e) {
        std::cout << "Error: " << e.what() << "\n";
        return 1;
    }
    return 0;
}
/*Exceptions separate normal logic from error handling but may introduce runtime overhead and complexity.*/

File: code/Rust/error_handling.rs
// Rust: error handling via Result
fn parse_positive(x: i32) -> Result<i32, String> {
    if x <= 0 {
        Err("input must be positive".into())
    } else {
        Ok(x)
    }
}

fn main() {
    match parse_positive(-3) {
        Ok(v) => println!("ok: {}", v),
        Err(e) => {
            println!("Error: {}", e);
            std::process::exit(1);
        }
    }
}
/*  Errors must be explicitly handled.
The compiler prevents ignoring failure cases. */